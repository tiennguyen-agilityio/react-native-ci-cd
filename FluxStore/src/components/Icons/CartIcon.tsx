import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const CartIcon = memo(
  ({
    width = 20,
    height = 20,
    disabled = false,
    isActive = false,
    color,
    style,
    onPress,
    ...props
  }: IconProps) => {
    const {
      theme: {text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="cart-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}
        {...props}>
        <Svg width={width} height={height} viewBox="0 0 20 20" fill="none">
          <Path
            d="M5 5H3.207A3 3 0 00.213 8.2l.6 9A3 3 0 003.807 20h12.386a3 3 0 002.994-2.8l.6-9A3 3 0 0016.793 5H15v2a1 1 0 11-2 0V5H7v2a1 1 0 01-2 0V5z"
            fill={color || isActive ? text.default : text.light}
          />
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M5 5a5 5 0 0110 0h-2a3 3 0 10-6 0H5z"
            fill={color || isActive ? text.default : text.light}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
