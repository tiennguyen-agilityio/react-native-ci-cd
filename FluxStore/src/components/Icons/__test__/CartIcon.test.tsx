import {render, fireEvent, screen} from '@testing-library/react-native';

import {CartIcon} from '../CartIcon';

describe('CartIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<CartIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive true', () => {
    const {toJSON} = render(<CartIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<CartIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('cart-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<CartIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('cart-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
