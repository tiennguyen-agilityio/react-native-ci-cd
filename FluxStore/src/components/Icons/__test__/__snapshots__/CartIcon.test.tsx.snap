// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="cart-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={20}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={20}
    vbWidth={20}
    width={20}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M5 5H3.207A3 3 0 00.213 8.2l.6 9A3 3 0 003.807 20h12.386a3 3 0 002.994-2.8l.6-9A3 3 0 0016.793 5H15v2a1 1 0 11-2 0V5H7v2a1 1 0 01-2 0V5z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        clipRule={0}
        d="M5 5a5 5 0 0110 0h-2a3 3 0 10-6 0H5z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`CartIcon renders correctly (matches snapshot) with isActive true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="cart-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={20}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={20}
    vbWidth={20}
    width={20}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M5 5H3.207A3 3 0 00.213 8.2l.6 9A3 3 0 003.807 20h12.386a3 3 0 002.994-2.8l.6-9A3 3 0 0016.793 5H15v2a1 1 0 11-2 0V5H7v2a1 1 0 01-2 0V5z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        clipRule={0}
        d="M5 5a5 5 0 0110 0h-2a3 3 0 10-6 0H5z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
